src/common.jl

```diff
@@ -1,225 +1,49 @@
-@grad function nextfloat(r::Real)
-    nextfloat(data(r)), Δ -> Δ
-end
-
-for f = [:hcat, :vcat]
-    for c = [
-        [:TrackedReal],
-        [:AbstractVecOrMat, :TrackedReal],
-        [:TrackedVecOrMat, :TrackedReal],
-    ]
-        cnames = map(_ -> gensym(), c)
-        @eval begin
-            function Base.$f(
-                $([:($x::$c) for (x, c) in zip(cnames, c)]...),
-                x::Union{TrackedArray,TrackedReal},
-                xs::Union{AbstractArray,Number}...,
-            )
-                return track($f, $(cnames...), x, xs...)
-            end
-        end
-    end
-    @eval begin
-        @grad function $f(x::Real)
-            $f(data(x)), (Δ) -> (Δ[1],)
-        end
-        @grad function $f(x1::Real, x2::Real)
-            $f(data(x1), data(x2)), (Δ) -> (Δ[1], Δ[2])
-        end
-        @grad function $f(x1::AbstractVector, x2::Real)
-            $f(data(x1), data(x2)), (Δ) -> (Δ[1:length(x1)], Δ[length(x1)+1])
-        end
-    end
-end
-
-function Base.copy(
-    A::TrackedArray{T, 2, <:Adjoint{T, <:AbstractTriangular{T, <:AbstractMatrix{T}}}},
-) where {T <: Real}
-    return track(copy, A)
-end
-@grad function Base.copy(
-    A::TrackedArray{T, 2, <:Adjoint{T, <:AbstractTriangular{T, <:AbstractMatrix{T}}}},
-) where {T <: Real}
-    return copy(data(A)), ∇ -> (copy(∇),)
-end
-
-Base.:*(A::TrackedMatrix, B::AbstractTriangular) = track(*, A, B)
-Base.:*(A::AbstractTriangular{T}, B::TrackedVector) where {T} = track(*, A, B)
-Base.:*(A::AbstractTriangular{T}, B::TrackedMatrix) where {T} = track(*, A, B)
-Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::TrackedMatrix) where {T} = track(*, A, B)
-Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::TrackedVector) where {T} = track(*, A, B)
-
-function Base.fill(
-    value::TrackedReal,
-    dims::Vararg{Union{Integer, AbstractUnitRange}},
-)
-    return track(fill, value, dims...)
-end
-@grad function Base.fill(value::Real, dims...)
-    return fill(data(value), dims...), function(Δ)
-        size(Δ) ≢  dims && error("Dimension mismatch")
-        return (sum(Δ), map(_->nothing, dims)...)
-    end
-end
-
-## StatsFuns ##
-
-logsumexp(x::TrackedArray; dims=:) = _logsumexp(x, dims)
-_logsumexp(x::TrackedArray, dims=:) = track(_logsumexp, x, dims)
-@grad function _logsumexp(x::TrackedArray, dims)
-    lse = logsumexp(data(x), dims = dims)
-    return lse, Δ -> (Δ .* exp.(x .- lse), nothing)
+function turing_chol(A::AbstractMatrix, check)
+    chol = cholesky(A, check=check)
+    (chol.factors, chol.info)
 end
-
-## Linear algebra ##
-
-# Work around https://github.com/FluxML/Tracker.jl/pull/9#issuecomment-480051767
-
-upper(A::AbstractMatrix) = UpperTriangular(A)
-lower(A::AbstractMatrix) = LowerTriangular(A)
-function upper(C::Cholesky)
-    if C.uplo == 'U'
-        return upper(C.factors)
-    else
-        return copy(lower(C.factors)')
+function ChainRules.rrule(::typeof(turing_chol), A::AbstractMatrix, check)
```

Comment:
Currently `chol_blocked_rev` is only tested for real matrices. Probably the right thing to do is relax that constraint on ChainRules and test (I'll open a PR).